azure/scope/cluster.go

```diff
@@ -711,33 +711,37 @@ func (s *ClusterScope) GetPrivateDNSZoneName() string {
 }
 
 // APIServerLBPoolName returns the API Server LB backend pool name.
-func (s *ClusterScope) APIServerLBPoolName(loadBalancerName string) string {
-	return azure.GenerateBackendAddressPoolName(loadBalancerName)
+func (s *ClusterScope) APIServerLBPoolName() string {
+	return s.APIServerLB().BackendPool.Name
 }
 
-// OutboundLBName returns the name of the outbound LB.
-func (s *ClusterScope) OutboundLBName(role string) string {
+// OutboundLB returns the outbound LB.
+func (s *ClusterScope) OutboundLB(role string) *infrav1.LoadBalancerSpec {
```

Comment:
- Are we using `OutboundLB()` anywhere else outside this package? Any specific reason on exporting it out?

Reply:
- If it's just the `lb.Name` we are interested in, can we just return the name instead of the whole spec?

Reply:
You're right should be private. I have decomposed the load balancer selection logic into a function because it is needed in both OutboundLBName and OutboundPoolName.